### DISCUSSION TITLE
Supporting of params surrounded by braces? ex: {varName}

### DISCUSSION THREAD

**Discussion by badloop** [sql-formatter-org/sql-formatter#551] [Q&A]
Python uses the brace format in its format strings and it is an easy way to format sql using keyword arguments.  Based on what I'm seeing in paramTypes there doesn't appear to be a way to support handling these during formatting.  

Is this supported and I'm just missing it?

**Comment by nene** [comment]
Currently not supported.

I'm not sure whether this is the same feature as discussed in #545 or whether this is a different kind of formatting system also found in Python. Either way, implementing such custom parameter format, as proposed there, should solve this.

**Reply by badloop** [reply]
Very nice! So if I'm understanding correctly, since I'm using a config file, my setup would be something like this:
```json
{
    "keywordCase": "upper",
    "tabWidth": 4,
    "commaPosition": "before",
    "language": "snowflake",
    "paramTypes": {
        "custom": [
            { "regex": "{.*}"}
        ]
    }
}
```
However when using the above and trying to process something like 
```sql
select * from {env}.schema.table;
```
I'm still getting an error:
`Error: Parse error: Unexpected "{env}.schema"`

**Reply by nene** [reply]
Well, that's because this feature is yet to be implemented :)

**Reply by badloop** [reply]
HAHA that makes sense.  Got a little eager I guess.  Is there a timeline for that? Just asking cause I actively need it but will need to work around if its going to be a while.  Really appreciate the help, btw!

**Reply by badloop** [reply]
Or even a temporary dev branch would work for me.

**Reply by nene** [reply]
Not sure. This feature request has been sitting there for about a month now. Though it's largely so because the author there hinted he might do a PR... but apparently it's not happening.

Depends on how motivated I am to actually implement it. It should be a relatively straightforward thing. Might need to minimally reorganize some code.

More people asking for a feature definitely helps.

**Reply by sonnguyen1410** [reply]
is there any update on this feature?

**Reply by nene** [reply]
No progress at the moment. Though I might take this on now that there's been 3 people asking for this.

**Reply by nene** [reply]
This is now implemented and released as 12.2.0.

See the [paramTypes](https://github.com/sql-formatter-org/sql-formatter/blob/master/docs/paramTypes.md#custom-parameter-syntax) docs.
